bigquery/cloud-client/async_query_params.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Command-line application to perform asynchronous queries in BigQuery.
+
+For more information, see the README.md under /bigquery.
+
+Example invocation:
+    $ python async_query_params.py 'romeoandjuliet' 100
+"""
+
+import argparse
+import time
+import uuid
+
+from google.cloud import bigquery
+
+
+def wait_for_job(job):
+    while True:
+        job.reload()  # Refreshes the state via a GET request.
+        if job.state == 'DONE':
+            if job.error_result:
+                raise RuntimeError(job.errors)
+            return
+        time.sleep(1)
+
+
+def async_query_params(corpus, min_word_count):
+    client = bigquery.Client()
+    query_job = client.run_async_query(
+        str(uuid.uuid4()),
+        """SELECT word, word_count
+        FROM `bigquery-public-data.samples.shakespeare`
+        WHERE corpus = ?
+        AND word_count >= ?
+        ORDER BY word_count DESC;
+        """,
+        query_parameters=(
+            bigquery.ScalarQueryParameter(
+                # Set the name to None to use positional parameters (? symbol
+                # in the query).  Note that you cannot mix named and positional
+                # parameters.
+                None,
+                'STRING',
+                corpus),
+            bigquery.ScalarQueryParameter(None, 'INT64', min_word_count)))
+
+    # Only standard SQL syntax supports parameters in queries.
+    # See: https://cloud.google.com/bigquery/sql-reference/
+    query_job.use_legacy_sql = False
+    query_job.begin()
+    wait_for_job(query_job)
+    print('Positional parameter query completed')
+
+    query_job = client.run_async_query(
```

Comment:
Done.

Yes, splitting into separate functions should make it a little easier to talk about in the docs I write for this, too.